### PR TITLE
[DOCS] Add Virtual Ref Documentation and tutorial

### LINKED ISSUE #214: [DOCS] Virtual datasets
We have a placeholder page on virtual datasets

https://github.com/earth-mover/icechunk/blob/fb5022199461e3b5b6aed22bcec132e470f3cc43/docs/docs/icechunk-python/virtual.md?plain=1#L1-L3

This needs to be filled in with wherever we are on virtual datasets. It's fine to put a bunch of warnings that this is all experimental, etc.

### PR DESCRIPTION
**Pull request by mpiannucci**
closes #214

### CODE REVIEW

File: docs/docs/icechunk-python/virtual.md [hunk 1]
```diff
@@ -1,3 +1,145 @@
...
+    config=StoreConfig(
+        virtual_ref_config=VirtualRefConfig.s3_anonymous(region='us-east-1'),
+    )
+)
+```
+
+With the store created, lets write our virtual dataset to Icechunk with VirtualiZarr!
+
+```python
+dataset_to_icechunk(virtual_ds, store)
```
[the remark this answers is not in the document]
Comment (mpiannucci) [reply docs/docs/icechunk-python/virtual.md:107]: Agree but we can update that later

File: docs/docs/icechunk-python/virtual.md [hunk 2]
```diff
@@ -1,3 +1,150 @@
 # Virtual Datasets
 
-Kerchunk, VirtualiZarr, etc.
\ No newline at end of file
+While Icechunk works wonderfully with native chunks managed by zarr, there are many times where creating a dataset relies on existing archived data. To allow this, Icechunk supports "Virtual" chunks, where any number of chunks in a given dataset may reference external data in existing archival formats, such as netCDF, HDF, GRIB, or TIFF. Virtual chunks simply load the raw data from the source data's original location without copying. moving, or modifying the original data files. This allows for using Icechunk to manage large datasets from existing data without needing that data to be in `zarr` format.
```
Comment (rabernat) [inline docs/docs/icechunk-python/virtual.md:3]:
```suggestion
While Icechunk works wonderfully with native chunks managed by Zarr, there is lots of archival data out there in other formats already. To interoperate with such data, Icechunk supports "Virtual" chunks, where any number of chunks in a given dataset may reference external data in existing archival formats, such as netCDF, HDF, GRIB, or TIFF. Virtual chunks are loaded directly from the original source without copying or modifying the original achival data files. This enables Icechunk to manage large datasets from existing data without needing that data to be in Zarr format already.
```

File: docs/docs/icechunk-python/virtual.md [hunk 3]
```diff
@@ -1,3 +1,150 @@
...
+
+!!! warning
+
+    While virtual references are fully supported in Icechunk, creating virtual datasets relies on using experimental or pre-release versions of open source tools. For full instructions on how to install the required tools and ther current statuses [see the tracking issue on Github](https://github.com/earth-mover/icechunk/issues/197).
```
Comment (rabernat) [inline docs/docs/icechunk-python/virtual.md:7]:
```suggestion
    While virtual references are fully supported in Icechunk, creating virtual datasets currently relies on using experimental or pre-release versions of open source tools. For full instructions on how to install the required tools and ther current statuses [see the tracking issue on Github](https://github.com/earth-mover/icechunk/issues/197).
    With time, these experimental features will make their way into the released packages.
```

File: docs/docs/icechunk-python/virtual.md [hunk 4]
```diff
@@ -1,3 +1,150 @@
...
+
+To create virtual Icechunk datasets with python, the community utilizes the [kerchunk](https://fsspec.github.io/kerchunk/) and [VirtualiZarr](https://virtualizarr.readthedocs.io/en/latest/) packages.
```
Comment (rabernat) [inline docs/docs/icechunk-python/virtual.md:9]:
```suggestion
To create virtual Icechunk datasets with Python, the community utilizes the [kerchunk](https://fsspec.github.io/kerchunk/) and [VirtualiZarr](https://virtualizarr.readthedocs.io/en/latest/) packages. 
```

File: docs/docs/icechunk-python/virtual.md [hunk 5]
```diff
@@ -1,3 +1,150 @@
...
+
+`VirtualiZarr` lets users ingest existing data files into virtual datasets using various different tools under the hood, including `kerchunk`, `xarray`, `zarr`, and now `icechunk`. It does so by creating virtual references to existing data that can be combined and manipulated to create larger virtual datasets using `xarray`. These datasets can then be exported to `kerchunk` reference format or to an `Icechunk` store, without ever copying or moving the existing data files.
+
+## Creating a virtual dataset with VirtualiZarr
+
+We are going to create a virtual dataset pointing to all of the [OISST](https://www.ncei.noaa.gov/products/optimum-interpolation-sst) data for August 2024. This data is distributed publicly as netCDF files on AWS S3, with one netCDF file containing the Sea Surface Temperature (SST) data for each day of the month. We are going to use `VirtualiZarr` to combine all of these files into a single virtual dataset spanning the entire month, then write that dataset to Icechunk for use in analysis.
+
+!!! note
+
+    At this point you should have followed the instructions [here](https://github.com/earth-mover/icechunk/issues/197) to install the necessary tools.
```
Comment (rabernat) [inline docs/docs/icechunk-python/virtual.md:21]:
```suggestion
    At this point you should have followed the instructions [here](https://github.com/earth-mover/icechunk/issues/197) to install the necessary experimental dependencies.
```

**Review by rabernat — APPROVED** [review]
Great!

A few minor wording changes, then please merge.

**Pull request merged**